### PR TITLE
initialize: use gphome instead of bindir

### PR DESCRIPTION
**Pull request by kalensk**
This more closely follows Greenplum conventions and standards, and allows us to avoid using relative directories to source greenplum_path. A test has been added to ensure the path is sanitized.

If needed, followup PRs can be made to further cleanup before closing out the story.

[Pipeline](https://cm.ci.gpdb.pivotal.io/teams/main/pipelines/gpupgrade:gphome)

### CODE REVIEW

File: gpupgrade_config [hunk 1]
```diff
@@ -7,16 +7,16 @@
...
-source_bindir =
+# The installation path for the source Greenplum Database.
+# For example, /usr/local/<source-greenplum-version>.
+# This parameter is required.
+source_gphome =
 
 
-# The path to the binary directory for the target Greenplum installation.
-# For example, /usr/local/<target-greenplum-version>/bin.
+# The installation path for the source Greenplum Database.
```
Comment (bhuvnesh2703) [inline gpupgrade_config:16]: should this be target?

File: cli/commands/commands.go [hunk 2]
```diff
@@ -264,8 +265,8 @@ func createConfigSetSubcommand() *cobra.Command {
 		},
 	}
 
-	subSet.Flags().String("source-bindir", "", "install directory for source gpdb version")
-	subSet.Flags().String("target-bindir", "", "install directory for target gpdb version")
+	subSet.Flags().String("source-gphome", "", "the install path for the source Greenplum installation")
```
Comment (bhuvnesh2703) [inline cli/commands/commands.go:268]: should this be `the path for the source Greenplum binaries installation` 
somehow the `install` path and `installation` sounds a bit off..

**Comment by kalensk** [comment]
I intentionally did not change most of the test parameters such as those in the existing BATs tests. But it sounds like those are important to you. I'll update them.

File: README.md [hunk 3]
```diff
@@ -83,7 +83,7 @@ make install # installs gpupgrade into $GOBIN
 ### Running
 
 ```
-gpupgrade initialize --source-bindir "$GPHOME/bin" --target-bindir "$GPHOME/bin" --source-master-port 6000 --disk-free-ratio 0
+gpupgrade initialize --source-gphome "$GPHOME" --target-gphome "$GPHOME" --source-master-port 6000 --disk-free-ratio 0
```
[the remark this answers is not in the document]
Comment (kalensk) [reply README.md:86]: Keep in mind that developers can install GPDB in any prefix, and `GPHOME_SOURCE` and `GPHOME_TARGET` are our conventions and not really known for new users who are reading the `README`. My intention when giving that example is to give new devs something that should mostly "just work" to get them up and running quickly. As they start to use the system my thought is they will learn quickly learn more.

File: hub/init_target_cluster.go [hunk 4]
```diff
@@ -161,7 +159,7 @@ func RunInitsystemForTargetCluster(stream step.OutStreams, target *greenplum.Clu
 	}
 
 	script := fmt.Sprintf("source %[1]s/greenplum_path.sh && %[1]s/bin/gpinitsystem %[2]s",
```
[the remark this answers is not in the document]
Comment (kalensk) [reply hub/init_target_cluster.go:161]: I intentionally left this as is, for now at least.

File: gpupgrade_config (hunk 1 above)
Comment (kalensk) [reply gpupgrade_config]: Fixed. Thank you!

(When working on a PR, after awhile my eyes start to glaze over. heh)

File: cli/commands/commands.go (hunk 2 above)
Comment (kalensk) [reply cli/commands/commands.go]: Updated based on PM and designer's input.

File: cli/commands/commands.go [hunk 5]
```diff
@@ -315,8 +316,8 @@ func createConfigShowSubcommand() *cobra.Command {
 	}
 
 	subShow.Flags().Bool("id", false, "show upgrade identifier")
-	subShow.Flags().Bool("source-bindir", false, "show install directory for source gpdb version")
-	subShow.Flags().Bool("target-bindir", false, "show install directory for target gpdb version")
+	subShow.Flags().Bool("source-gphome", false, "show the install path for the source Greenplum installation")
```
[the remark this answers is not in the document]
Comment (kalensk) [reply cli/commands/commands.go]: A single comment with "update all references to the following..." or similar should be sufficient.

File: cli/commands/commands.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kalensk) [reply cli/commands/commands.go]: Updated on PM and designer's input.

**Pull request merged**